streambed-logged/src/compaction.rs

```diff
@@ -239,16 +239,20 @@ where
     E: Error,
     W: Write,
 {
-    pub fn new<AA, NTF, RHFWWF>(
+    pub fn new<AA, NWF, RCHF, RPHF>(
         age_active: AA,
-        new_work_file: NTF,
-        replace_history_files: RHFWWF,
+        new_work_file: NWF,
+        mut recover_history_files: RCHF,
+        replace_history_files: RPHF,
     ) -> Self
     where
         AA: FnMut() -> Result<Option<Offset>, E> + Send + 'static,
-        NTF: FnMut() -> Result<W, E> + Send + 'static,
-        RHFWWF: FnMut() -> Result<(), E> + Send + 'static,
+        NWF: FnMut() -> Result<W, E> + Send + 'static,
+        RCHF: FnMut() -> Result<(), E> + Send + 'static,
+        RPHF: FnMut() -> Result<(), E> + Send + 'static,
     {
+        let _ = recover_history_files();
```

Comment:
This is the new functionality. When compaction is initialised, a recovery of history files is performed where anything that was "in progress" from a previous run is removed. Compaction may then start with in a clean state.